plugins/woocommerce/src/Admin/Features/LaunchYourStore.php

```diff
@@ -93,4 +96,38 @@ public function preload_settings( $settings ) {
 
 		return $settings;
 	}
+
+	/**
+	 * Add 'coming soon' banner on the frontend when the following conditions met.
+	 *
+	 * - User must be either an admin or store editor (must be logged in).
+	 * - 'woocommerce_coming_soon' option value must be 'yes'
+	 */
+	public function maybe_add_coming_soon_banner_on_frontend() {
+		// User must be an admin or editor.
+		// phpcs:ignore
+		if ( ! current_user_can( 'editor' ) && ! current_user_can( 'administrator' ) ) {
+			return false;
+		}
+
+		// 'woocommerce_coming_soon' must be 'yes'
+		if ( get_option( 'woocommerce_coming_soon', 'no' ) !== 'yes' ) {
+			return false;
+		}
+
+		$link = admin_url( 'admin.php?page=wc-settings#wc_settings_general_site_visibility_slotfill' );
+
+		$text = sprintf(
+			// translators: no need to translate it. It's a link.
+			__(
+				"
+			This page is in \"Coming soon\" mode and is only visible to you and those who have permission. To make it public to everyone,&nbsp;<a href='%s'>change visibility settings</a>.
+		",
+				'woocommerce'
+			),
+			$link
+		);
+		// phpcs:ignore
+		echo "<div id='coming-soon-footer-banner'>$text</div>";
```

Comment:
There needs to be a dismiss button as well, which should close the banner for the current session. You might need to sprinkle in some JS to do this.

Reply:
My apologies! I missed that from Figma 🤦 

> The dismiss button could be done in a follow-up.

Sounds good!